tests/test_copies.py

```diff
@@ -0,0 +1,201 @@
+    assert df_test_1.equals(df_test_3) is False
+    np.testing.assert_almost_equal(df_test_1["feat_1"].mean(), 0, decimal=0)
+    np.testing.assert_almost_equal(df_test_1["feat_2"].mean(), 20, decimal=0)
+    np.testing.assert_almost_equal(df_test_1["feat_1"].std(), 2, decimal=0)
+    np.testing.assert_almost_equal(df_test_1["feat_2"].std(), 40, decimal=0)
+
+
+def test_labeling():
+    train_dataset = pd.DataFrame(
+        np.random.rand(4, 3) * 80 + np.tile(np.array([10, 40, 130]), (4, 1))
+    )
+    original_classifier = DummyClassifier(strategy="constant", constant="dummy_class")
+    original_classifier.fit(train_dataset, np.array(["a", "b", "dummy_class", "a"]))
+
+    unlabeled_dataset = pd.DataFrame(
+        np.random.rand(4, 3) * 80 + np.tile(np.array([10, 40, 130]), (4, 1))
+    )
+    label_col = "potato"
+    df_labeled = labeling(unlabeled_dataset, original_classifier, label_col=label_col)
+
+    assert isinstance(df_labeled, Dataset) is True
+    assert len(df_labeled.df["potato"]) == 4
+    assert df_labeled.df["potato"].unique() == "dummy_class"
+
+
+def test_empirical_fidelity_error():
+    y_pred_original = [1, 0, 1, 0]
+    y_pred_copy1 = [1, 1, 0, 0]
+    y_pred_copy2 = [1, 0, 0, 0]
+    efe1 = empirical_fidelity_error(y_pred_original, y_pred_copy1)
+    efe2 = empirical_fidelity_error(y_pred_original, y_pred_copy2)
+    assert efe1 == 0.5
+    assert efe2 == 0.25
+
+
+def test_ClassifierCopy_copy_classifier():
+    # Original classifier
+    train_data = pd.DataFrame(
+        {"x": [0, 1, 0, 2, 1], "y": [1, 0, 2, 0, 1], "label": [0, 0, 1, 1, 1]},
+        columns=["x", "y", "label"],
+    )
+    original_classifier = SVC(kernel="linear", random_state=42)
+    original_classifier.fit(train_data[["x", "y"]], train_data["label"])
+
+    # Copy classifier
+    feature_parameters = {"x": {"min": 0, "max": 2}, "y": {"min": 0, "max": 2}}
+    classifier_copy = DecisionTreeClassifier(max_depth=2, random_state=42)
+    copy_grid = ClassifierCopy(
+        original_classifier,
+        classifier_copy,
+        grid_sampling,
+        nsamples=900,
+        label_col="label",
+        feature_parameters=feature_parameters,
+    )
+    train_data_copy = copy_grid.copy_classifier(get_training_data=True)
+
+    assert isinstance(train_data_copy, Dataset) is True
+    assert len(train_data_copy.df["label"]) == 900
+    assert list(train_data_copy.df["label"].unique()) == [0, 1]
+
+    # Evaluate classifier
+    test_data = pd.DataFrame(
+        {"x": [-1, 0, 1, 2], "y": [-1, 0, 1, 2]}, columns=["x", "y"]
+    )
+    efe = copy_grid.compute_fidelity_error(test_data=test_data)
+
+    assert efe == 0
+
+
+def test_multiclass_gaussians():
+    nsamples = 100
+    nclasses = 15
+    nfeatures = 30
+    df_2feat_2class = multiclass_gaussians(
+        nsamples=nsamples,
+        nfeatures=nfeatures,
+        nclasses=nclasses,
+        center_low=2,
+        center_high=10,
+        scale_low=1,
+        scale_high=1,
+    ).df
```

Comment:
Maybe use non-default center/scale params, to make sure that code path is covered.